yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -83,10 +85,25 @@ export async function computeContractAddress(
   return AztecAddress.fromBuffer(resultBuf);
 }
 
-export async function computeContractLeaf(wasm: CircuitsWasm, leafPreimage: NullifierLeafPreimage) {
-  const data = leafPreimage.toBuffer();
-  wasm.call('pedersen__init');
-  wasm.writeMemory(0, leafPreimage.toBuffer());
-  await wasm.asyncCall('abis__compute_contract_leaf', 0, data.length);
-  return Buffer.from(wasm.getMemorySlice(data.length, data.length + 32));
+export function computeContractLeaf(wasm: CircuitsWasm, newContractData: NewContractData): Promise<Buffer> {
+  if (
+    newContractData.contractAddress.isZero() &&
+    newContractData.portalContractAddress.isZero() &&
+    newContractData.functionTreeRoot.isZero()
+  ) {
+    return Promise.resolve(Buffer.alloc(32, 0));
+  }
+  return Promise.resolve(
+    pedersenCompressInputs(wasm as any as BarretenbergWasm, [
+      newContractData.contractAddress.toBuffer(),
+      newContractData.portalContractAddress.toBuffer32(),
+      newContractData.functionTreeRoot.toBuffer(),
+    ]),
+  );
+  // const data = newContractData.toBuffer();
```

Comment:
Look like this doesn't have to be async now. And that will make the code that uses it cleaner!

Reply:
It's just a temporary fix, we want to replace it back with the old call when fixed on circuits